### PR TITLE
[W5.11][T09-A1] Fuad B Sahmawi

### PR DESCRIPTION
**Pull request by fuadsahmawi**
Sort command which helps sort address book by name in alphabetical order. Sorting is temporary as original internal list is not overwritten. Enhancement #2.

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -36,6 +36,16 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
         return String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, personsDisplayed.size());
     }
 
+    /**
```
Comment (kohyeeru) [inline src/seedu/addressbook/commands/Command.java:39]: Great that you included a header comment for the method!

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -81,6 +81,10 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Returns a modifiable java List view with elements cast as mutable {@link Person}s.
+     */
+    public List<Person> mutableListView() { return internalList;}
```
Comment (kohyeeru) [inline src/seedu/addressbook/data/person/UniquePersonList.java:87]: Just minor formatting issue; but it's missing a space after the colon in this line.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (kohyeeru) [inline src/seedu/addressbook/parser/Parser.java:14]: It's better to declare imports explicitly instead of using *, I think!

**Review by kohyeeru — COMMENTED** [review]
Great that commits were done in small incremental steps with easy to follow titles. However, take note that subject lines in commits should be written in imperative mood (e.g. add instead of added). Also, not sure how it happened but it seems like in between some commits, part of your subject line became the body; might want to be careful with that. :) See [here](https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message) for more info on the formats and convention.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Sorts the list of all persons in the address book by name in alphabetical order.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book as a list with index numbers sorted alphabetically by name.\n"
```
Comment (dalessr) [inline src/seedu/addressbook/commands/SortCommand.java:18]: Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer (not sure whether this line exceeds or not! :) )

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -81,6 +81,10 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Returns a modifiable java List view with elements cast as mutable {@link Person}s.
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:85]: The header comment does not follow the convention. All non trivial methods should have javadoc format header comments. (First line is the description of method, followed by @param for specifying the parameters and @return for specifying the return values, if applicable)

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 2 above)
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:87]: Coding standard violation. Layout/brace style problem.

File: src/seedu/addressbook/parser/Parser.java (hunk 3 above)
Comment (dalessr) [reply src/seedu/addressbook/parser/Parser.java:14]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.

This is a coding style violation, please avoid doing this in future.

**Review by dalessr — COMMENTED** [review]
Good job on covering all aspects of the new command @fuadsahmawi 👍

Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**